packages/frontend-2/components/project/page/discussions/Results.vue

```diff
@@ -39,61 +33,26 @@ const props = defineProps<{
   includeArchived: boolean
 }>()
 
-const logger = useLogger()
-
-const queryFilterVariables = computed(
-  (): ProjectCommentsFilter => ({
-    includeArchived: !!props.includeArchived
-  })
-)
-
-const infiniteLoaderId = ref('')
 const {
-  result: extraPagesResult,
-  fetchMore: fetchMorePages,
-  variables: resultVariables,
-  onResult
-} = useQuery(latestCommentThreadsQuery, () => ({
-  projectId: props.project.id,
-  filter: queryFilterVariables.value
-}))
+  identifier,
+  onInfiniteLoad,
+  query: { result }
+} = usePaginatedQuery({
+  query: latestCommentThreadsQuery,
+  baseVariables: computed(() => ({
+    projectId: props.project.id,
+    filter: { includeArchived: !!props.includeArchived }
+  })),
+  resolveKey: (vars) => [vars.projectId],
```

Comment:
Just the projectId isn't enough - vars.filter being different will also result in a different query. If you leave vars.filter outside of the key, a change to the filters won't reset the pagination state, a new query won't be invoked etc.

Reply:
Good point, updated it :)